### DISCUSSION TITLE
Feedback using `winnow` as a replacement for `nom` in `jql`

### DISCUSSION THREAD

**Discussion by yamafaktory** [winnow-rs/winnow#385] [Show and tell]
This is basically a follow-up discussion of the initial PR I made to switch from `nom` to `winnow` after getting some good feedback on Mastodon.

I'd like to thanks @epage for taking the time to jump into this initial PR, providing very good hints on how to better use the capabilities of the library - and improving performance.

I'm not going to dive into details since most of the changes were pretty straightforward based on the direct feedback. What I really liked was:
- `Parser` being implemented for strings and byte strings out of the box (no need to use `tag` most of the time)
- Cleaner methods like `separated` directly taking a `range` as first argument
- Having a clean `Parser` trait compared to `nom` which has more cumbersome types
- Easy way to parse a `str` to some number with e.g. `digit1.parse_to().parse_next(input)`
- More natural way of calling `map` on a `Parser` compared to `nom` where it's a another combinator
-  `repeat` as a way to accumulate the output of a parser into a `Vec`

What would be very beneficial would be to have an official how-to for users migrating from `nom` to `winnow`, condensing all the examples from:
- https://github.com/gitext-rs/git-config-env/pull/11
- https://github.com/crate-ci/git-conventional/pull/37
- https://github.com/crate-ci/typos/pull/664
- https://github.com/Byron/gitoxide/pulls?q=is%3Apr+winnow+is%3Aclosed

**Comment by epage** [comment]
Thanks for the feedback!

#387 adds a migration guide

**Reply by yamafaktory** [reply]
Thanks for the swift addition!
